commitizen/cz/base.py

```diff
@@ -27,7 +27,14 @@ class BaseCommitizen(metaclass=ABCMeta):
     # It can be modified per rule
     commit_parser: Optional[str] = r"(?P<message>.*)"
     changelog_pattern: Optional[str] = r".*"
-    changelog_map: Optional[dict] = None  # TODO: Use it
+    change_type_map: Optional[dict] = None
+
+    # Executed per message parsed by the commitizen
+    message_hook: Optional[Callable] = None  # (dict, GitCommit) -> dict
```

Comment:
Maybe we could use annotation like `Optional[Callable[[dict, GitCommit], dict]`

Reply:
Cool, much better actually haha. My intention was there with the comment :rofl: